File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque
input = sys.stdin.readline
com = int(input())
dic = {i:[] for i in range(1, com+1)}
visit = [0]*(com+1)
visit[1] = 1
result = 0
for _ in range(int(input())):
    s,e = map(int,input().split())
    dic[s].append(e)
    dic[e].append(s)

q = deque([1])

while q:
    s = q.popleft()
    for i in dic[s]:
        if not visit[i]:
            visit[i] = 1
            result +=1
            q.append(i)
print(result)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;
import java.util.stream.*;
public class Main {
    static List<List<Integer>> nodes;
    static boolean[] visit;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int net = Integer.parseInt(br.readLine());

        nodes = IntStream.rangeClosed(0,N).mapToObj(ArrayList<Integer>::new).collect(Collectors.toList());

        for (int i=0; i<net; i++) {
            int[] node = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            nodes.get(node[0]).add(node[1]);
            nodes.get(node[1]).add(node[0]);
        }

        visit = new boolean[N+1];

        // bfs
        Deque<Integer> queue = new ArrayDeque<>(){{
            add(1);
        }};

        System.out.println(bfs(queue));

    }

    public static int bfs(Deque<Integer> queue){
        int cnt = 0;
        while (!queue.isEmpty()) {
            int now = queue.pollFirst();
            if (!visit[now]) visit[now] = true;

            for (int num: nodes.get(now)) {
                if (!visit[num]) {
                    queue.add(num);
                    visit[num] = true;
                    cnt++;
                }
            }
        }
        return cnt;
    }
}
